interface/modules/zend_modules/config/autoload/global.php

```diff
@@ -9,41 +9,96 @@
  * @NOTE: In practice, this file will typically be INCLUDED in your source
  * control, so do not include passwords or other sensitive information in this
  * file.
+ *
+ * @Author: Oshri R <oshri.rozmarin@gmail.com>
+ *
  */
 
-if ($GLOBALS['disable_utf8_flag']) {
-  $db = array(
-        'driver'         => 'Pdo',
-        'dsn'            => 'mysql:dbname='.$GLOBALS['dbase'].';host='.$GLOBALS['host'],
-        'username'       => $GLOBALS['login'],
-        'password'       => $GLOBALS['pass'],
-        'port'           => $GLOBALS['port'],
-        'driver_options' => array(
-            PDO::MYSQL_ATTR_INIT_COMMAND => 'SET sql_mode = \'\''
-        )
-    );
-} else {
-  $db = array(
+// Sets default factory using the default database
+$factories = array(
+    'Zend\Db\Adapter\Adapter' => function ($serviceManager) {
+        $adapterFactory = new Zend\Db\Adapter\AdapterServiceFactory();
+        $adapter = $adapterFactory->createService($serviceManager);
+        \Zend\Db\TableGateway\Feature\GlobalAdapterFeature::setStaticAdapter($adapter);
+        return $adapter;
+    }
+);
+
+// Open pdo connection
+$dbh = new PDO('mysql:dbname=' . $GLOBALS['dbase'] . ';host=' . $GLOBALS['host'], $GLOBALS['login'], $GLOBALS['pass']);
+$adapters = array();
+$res = $dbh->prepare('SELECT * FROM zf_multiple_db');
+if($res->execute()){
+    foreach ($res->fetchAll() as $row) {
+
+        // Create new adapters using data from database
+        $adapters[$row['namespace']] = array(
+            'driver' => 'Pdo',
+            'dsn' => 'mysql:dbname=' . $row['dbname'] . ';host=' . $row['host'] . '',
+            'driver_options' => array(
+                PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES \'UTF8\''
+            ),
+            'port' => $row['port'],
+            'username' => $row['username'],
+            'password' => my_decrypt($row['password']),
+        );
+
+        // Create new factories using data from custom database
+        $factories[$row['namespace']] = function ($serviceManager) use ($row) {
+            $adapterAbstractServiceFactory = new Zend\Db\Adapter\AdapterAbstractServiceFactory();
+            $adapter = $adapterAbstractServiceFactory->createServiceWithName($serviceManager,'',$row['namespace']);
+            return $adapter;
+        };
+    }
+}
+$dbh = null; // Close pdo connection
+
+// If to use utf-8 or not in my sql query
+$utf8 =  ($GLOBALS['disable_utf8_flag']) ? array(PDO::MYSQL_ATTR_INIT_COMMAND => 'SET sql_mode = \'\'') : array(PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES \'UTF8\', sql_mode = \'\'');
```

Comment:
Recommend placing this line above the creation of the other adapters and using it there also for driver_options setting. Since those databases will also need to set utf8 and also clear the sql_mode.